compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/Token.java

```diff
@@ -82,9 +84,22 @@ public Token modify(MinutiaeList leadingMinutiae, MinutiaeList trailingMinutiae)
             return Collections::emptyIterator;
         }
 
-        return () -> internalNode.diagnostics().stream()
+        return () -> collectDiagnostics().iterator();
+    }
+
+    private List<Diagnostic> collectDiagnostics() {
+        List<Diagnostic> diagnosticList = new ArrayList<>();
+
+        // Collect diagnostics inside invalid token minutiae
+        leadingInvalidTokens().forEach(token -> token.diagnostics().forEach(diagnosticList::add));
+        trailingInvalidTokens().forEach(token -> token.diagnostics().forEach(diagnosticList::add));
+
+        // Collect token diagnostics
+        internalNode.diagnostics().stream()
```

Comment:
Should we use forEach instead of the stream? Since we are now replacing stream with foreach

Reply:
shall we stick the same way as the nonterminal node?
https://github.com/ballerina-platform/ballerina-lang/blob/a67edec2f67a3df1f966f4ae87396cd3aa2af817/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/NonTerminalNode.java#L305-L307

Reply:
Should we update that as well then? :)